### PR TITLE
Enable formating of the `network` module

### PR DESCRIPTION
**Pull request by tcharding**
The `network` module is not currently included in formatting. Also, at this point in time not much is happening in the `network` module so formatting it should not cause too many merge conflicts with other in-progress PRs.

The first 3 patches are formatting preparation of the repo, the next 2 are formatting preparation of the `network` module. The last patch is the result of running `cargo +nightly fmt`. Can one reviewer please verify the last patch consists only of `rustfmt` changes by running the command on their branch. Thanks

cc luckysori :)

### CODE REVIEW

File: bitcoin/src/network/message.rs [hunk 1]
```diff
@@ -672,7 +682,8 @@ mod test {
             0xfa, 0xa9, 0x95, 0x59, 0xcc, 0x68, 0xa1, 0xc1,
             0x10, 0x2f, 0x53, 0x61, 0x74, 0x6f, 0x73, 0x68,
             0x69, 0x3a, 0x30, 0x2e, 0x31, 0x37, 0x2e, 0x31,
-            0x2f, 0x93, 0x8c, 0x08, 0x00, 0x01, 0, 0 ];
+            0x2f, 0x93, 0x8c, 0x08, 0x00, 0x01, 0,    0
```
Comment (luckysori) [inline bitcoin/src/network/message.rs:795]: I think the trailing `0`s should be `0x00`s for readability.

**Review by luckysori — DISMISSED** [review]
Can confirm that both c6e96ec and e8dc4e2 are the result of running `cargo +nightly fmt`.

I ran [this command](https://stackoverflow.com/a/23527631) to verify this.

**Review by apoelstra — APPROVED** [review]
ACK c6e96ec02c804ef77039731d8b27846995e0e2e2

**Comment by apoelstra** [comment]
Are you aware of the rustfmt option `group_imports = StdExternalCrate`? Would this avoid the need to hackily reorder `mod` statements?

**Comment by dpc** [comment]
> Are you aware of the rustfmt option `group_imports = StdExternalCrate`? Would this avoid the need to hackily reorder `mod` statements?

We've went out of our way to use it in Fedimint (using stable for building, yet nightly for rustfmt) and it's a really nice option, IMO).

**Comment by tcharding** [comment]
> Are you aware of the rustfmt option `group_imports = StdExternalCrate`? Would this avoid the need to hackily reorder `mod` statements?

Yep we have that turned on, it because of that option I guess that the formatter wanted to put those to imports together and I wanted to keep them apart (the separation doesn't add that much but since it was trivial to force it I went for it).

File: bitcoin/src/network/message.rs (hunk 1 above)
Comment (tcharding) [reply bitcoin/src/network/message.rs:795]: Certainly - done, thanks man!

**Comment by tcharding** [comment]
Changes in force push:
- rebased to pick up fixes to `taproot-psbt`.
- Used `0x00` as suggested
- Re-ran formatter in patch 3 (some additional formatting issues were in `taproot-psbt`) 

No other changes.

**Review by apoelstra — DISMISSED** [review]
ACK d73c71f4e7e5848ee1410d00d359e1f37fafda8c

**Comment by tcharding** [comment]
Now includes an additional initial patch that improves the crate root re-exports and removes the import re-order kludge discussed above. CC @Kixunil since I had to manually touch the imports statements I went ahead and used your `pub extern crate` suggestion.

**Comment by tcharding** [comment]
Changes in force push:
- Fix CI by checking each patch and ensuring that my claims in the commit logs are correct after rebase :)

**Review by apoelstra — DISMISSED** [review]
ACK d2abd2b5acf0070b48dd24622dbdd19914daa6b9

**Review by apoelstra — DISMISSED** [review]
ACK 4d422cc5fb8ebaad3d9b1e9d6401e3f8e261e2bf

**Review by Kixunil — APPROVED** [review]

File: bitcoin/src/amount.rs [hunk 2]
```diff
@@ -413,13 +411,18 @@ fn fmt_satoshi_in(
     }
 
     let width = options.width.unwrap_or(0);
-    let (left_pad, pad_right) = match (num_width < width, options.sign_aware_zero_pad, options.align.unwrap_or(fmt::Alignment::Right)) {
+    let (left_pad, pad_right) = match (
+        num_width < width,
+        options.sign_aware_zero_pad,
+        options.align.unwrap_or(fmt::Alignment::Right),
```
[the remark this answers is not in the document]
Comment (tcharding) [reply bitcoin/src/amount.rs]: Awesome, it works. Added as an additional patch `1ecf0935 Add local variable to reduce line length`

**Review by Kixunil — APPROVED** [review]

**Comment by tcharding** [comment]
;) thanks @Kixunil

**Comment by apoelstra** [comment]
I should have an ACK in the next 20 minutes or so. This PR was long enough that it hit multiple bugs in my test script's "already done this check, don't fill up /tmp trying to redo it" logic ... and would therefore fill up /tmp and crash, every time. (The correct behavior is that it records all the checks that it managed to do, and avoids redoing them, so I can just run the whole command repeatedly and it'll eventually work.)

In particular, I learned that if you sign a cherry-picked commit with GPG, this randomizes the commit ID, so even if you screw around with the commit/author dates, you won't be able to get the same commit ID twice. I also learned that `git notes append -m` will strip trailing whitespace from whatever you give it, so if you foolishly add a string and expect to be able to find it later, you won't.

This kinda shit is why I always give evasive answers to people who ask me for my scripts..

**Review by apoelstra — APPROVED** [review]
ACK e04a7a926dfa5e943f1db3cec97eab9b19c10269

**Comment by Kixunil** [comment]
> I learned that if you sign a cherry-picked commit with GPG, this randomizes the commit ID

Sounds kinda surprising from you since git is literally a blockchain. ;)

**Comment by apoelstra** [comment]
>Sounds kinda surprising from you since git is literally a blockchain. ;)

What was surprising to me was that the gpg sig wasn't deterministic :)

**Comment by tcharding** [comment]
Lately I've been using `git commit --amend --no-edit` to change the commit ID to force CI to rerun, so I've been wondering what data changed. Using `git show --format=raw` as you mention does not give all the data because non of the output changes across the command above except the commit ID.
```
commit 6bf48962b3cafff6765f44b8fac38dfc828d77b4
tree 67e7c2a50665d2945f3afc576ca122030ee5b6ff
parent ad2e75e72c7d3dca6ecfaf0dddbbbdfc4737ef61
author Tobin Harding <me@tobin.cc> 1642971420 +1100
committer Tobin C. Harding <me@tobin.cc> 1668982614 +1100

commit 84fdc8ebaec0316e426f532c492b4dfda996a0e1
tree 67e7c2a50665d2945f3afc576ca122030ee5b6ff
parent ad2e75e72c7d3dca6ecfaf0dddbbbdfc4737ef61
author Tobin Harding <me@tobin.cc> 1642971420 +1100
committer Tobin C. Harding <me@tobin.cc> 1668982640 +1100
``` 

What else is going in there?

**Comment by Kixunil** [comment]
> then you can get the same commit ID

WTF? Are we talking about the same thing? Commits hash the hashes of previous ones so getting the same ID for two commits based on different things should be impossible.

**Comment by tcharding** [comment]
> Commits hash the hashes of previous ones so getting the same ID for two commits based on different things should be impossible.

I had guessed (I obviously do not know) that the commit hashed the hashes of previous ones _plus_ some time/date data so that each time we got a different hash. Hence why I was surprised that in the output of `cargo show --format=raw` the time/date was included but not changed?

**Comment by tcharding** [comment]
"1668982640 +1100" - that bit is the time and date info in some format, right? (I'm in +11 UTC)

**Comment by apoelstra** [comment]
>WTF? Are we talking about the same thing? Commits hash the hashes of previous ones so getting the same ID for two commits based on different things should be impossible.

I think we are talking about different things :). The commits I have which having matching IDs are identical in all respects, including their history.

>What else is going in there?

I'm surprised `format=raw` isn't showing you this, but almost certainly what's different is the commit date (which is distinct from the author date, which is what git normally shows).

**Comment by tcharding** [comment]
> I'm surprised format=raw isn't showing you this, but almost certainly what's different is the commit date (which is distinct from the author date, which is what git normally shows).

Ahh, sweet. Thanks

**Pull request merged**